ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallDialog.kt

```diff
@@ -39,16 +39,18 @@ fun PaywallDialog(
         }
     }
     if (shouldDisplayDialog) {
-        val dismissRequest = {
-            paywallDialogOptions.dismissRequest?.invoke()
-            shouldDisplayDialog = false
-        }
-
         Dialog(
-            onDismissRequest = dismissRequest,
+            onDismissRequest = {
+                shouldDisplayDialog = false
+                paywallDialogOptions.dismissRequest?.invoke()
+            },
             properties = DialogProperties(usePlatformDefaultWidth = shouldUsePlatformDefaultWidth()),
         ) {
-            DialogScaffold(paywallDialogOptions.toPaywallOptions(dismissRequest))
+            DialogScaffold(
+                paywallDialogOptions.toPaywallOptions(dismissRequest = {
+                    shouldDisplayDialog = false
+                }),
```

Comment:
I think this would be simpler keeping the `onDismissRequest` like it was before #1403:
```kotlin
dismissRequest = {
  shouldDisplayDialog = false
}
```

Then you can pass it here, and for the `Dialog` above just do:
```kotlin
onDismissRequest = {
  dismissRequest()
  paywallDialogOptions.dismissRequest?.invoke()
}
```

What do you think?